src/Form/DataField/LargeInt.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef XCSOAR_DATA_FIELD_LARGEINT_HPP
+#define XCSOAR_DATA_FIELD_LARGEINT_HPP
+
+#include "Number.hpp"
+#include "time/PeriodClock.hpp"
+
+class DataFieldLargeInt final : public NumberDataField
```

Comment:
But for this decision I need such a class. If XCSoar has a similar one I would be use, but I couldn't found.
And a list with 1000-10000 entries doesn't make any sense for me.

In a later usage I will pickup this (hidden) number with a (filtered) input of an user name (but this can only done with server connection  (with internet). This makes some additional work too...

Reply:
With 'AddValue' I should use only one function
```
 void AddValue(const char* name, const char* value);
```
and not the variants with uint32_t  and TCHAR* parameter? Or do you mean to remove only the TCHAR* parameter?

Reply:
> But for this decision I need such a class. If XCSoar has a similar one I would be use, but I couldn't found.

There is `DataFieldInteger`. You know it because you copied code from there.